packages/fulfillment/src/models/fulfillment-item.ts

```diff
@@ -100,10 +110,12 @@ export default class FulfillmentItem {
   @BeforeCreate()
   onCreate() {
     this.id = generateEntityId(this.id, "fulit")
+    this.fulfillment_id ??= this.fulfillment.id
   }
 
   @OnInit()
   onInit() {
     this.id = generateEntityId(this.id, "fulit")
+    this.fulfillment_id ??= this.fulfillment.id
```

Comment:
question: I guess this pattern will need to be applied in all modules?

Reply:
this is dependent on the relation you are creating, but if you are persisting from a parent then mikro orm assign to the relation and we transfert the id as we do not persist the relation

Reply:
yes definately, in some modules, the configuration is different and the repository and some service are managing the transposition, so yes some cleanup will be good for sure